### PR TITLE
Differenciate between roads and pedestrian streets

### PR DESCRIPTION
**Pull request by ENT8R**
While reviewing StreetComplete-related notes I saw many people irritated by the fact that StreetComplete can't differenciate between roads for cars and pedestrian streets.
This PR fixes this issue.

**Some examples are:**
https://www.openstreetmap.org/note/1296735
https://www.openstreetmap.org/note/1179245
https://www.openstreetmap.org/note/1174269
https://www.openstreetmap.org/note/1174268
https://www.openstreetmap.org/note/1174267
https://www.openstreetmap.org/note/1123039
https://www.openstreetmap.org/note/1290931
https://www.openstreetmap.org/note/1289562

### CODE REVIEW

**Comment by ENT8R** [comment]
I also added another answer for if the street is tagged with `area=yes` ("What is the name of this square?") as this was confusing to some users too...

**Comment by westnordost** [comment]
That's a very good idea, nice!

**Comment by ENT8R** [comment]
> The quest explicitly excludes any areas.

But what about this note: https://www.openstreetmap.org/note/1297034 which references https://www.openstreetmap.org/way/365260406 with `area=yes`...

**Comment by rugk** [comment]
and the last note is about the _surface quest_, not the name quest.

**Comment by rugk** [comment]
BTW, generally, just saw what I once wrote for that quest in the wiki as a question: "What is the name of this place?"
That can be asked for areas, streets or whatever… so that may be a way to make it more general.

**Comment by rugk** [comment]
Uh, wait, that's another quest… But if you are going into areas that is not really a road quest…

**Comment by matkoniecz** [comment]
Maybe it was done using version not including https://github.com/westnordost/StreetComplete/commit/10f8beb13f768b4f36295dce4d97e34d7a5074fd ?

**Comment by ENT8R** [comment]
Oops... It was the surface quest... I will revert the last commit.

**Comment by ENT8R** [comment]
I removed the additional area question from the road name question but added it to the road surface question as it might make sense to include it there...

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java [hunk 1]
```diff
@@ -164,7 +164,10 @@ else if(noProperRoad == AddRoadNameForm.IS_LINK)
 	@Override public String getCommitMessage() { return "Determine road names and types"; }
 	@Override public int getIcon() { return R.drawable.ic_quest_street_name; }
 	@Override public int getTitle() { return R.string.quest_streetName_title; }
-	@Override public int getTitle(@NonNull Map<String,String> tags) { return getTitle(); }
+	@Override public int getTitle(@NonNull Map<String,String> tags) {
+		if (tags.get("highway").equals("pedestrian")) return R.string.quest_streetName_pedestrian_title;
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java]: possible nullpointer

Comment (matkoniecz) [reply app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java]: Is it not enough to assume that thanks to https://github.com/ENT8R/StreetComplete/blob/c69465407a868021fb8f843ba9795a2fc47be3b4/app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java#L42 highway tag will be always present?

Comment (ENT8R) [reply app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java]: I thought that too...

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java]: Possibly, there won't be a problem with this now, but there may be in the future:

By contract (=by interface), it is only guaranteed that the `tags` parameter will never be null, it does not say anything about which tags must be present.

Note that the default implementation of `getTitle` in `SimpleOverpassQuestType` is:
```
@Override public final int getTitle()
{
	return getTitle(Collections.emptyMap());
}
```

So, asserting that every call to `getTitle(@NonNull Map<String,String> tags)` must carry the correct `tags` from the associated element with it, comes down to **asserting that `getTitle()` will never be called on `OsmQuestType`s**. Ulteriorly "removing" methods from the interface this way is prone to error, as this is not obvious to anyone.

**Comment by westnordost** [comment]
I'll just merge it and correct it afterwards.

**Pull request merged**